src/main/kotlin/com/bantolomeus/service/BookService.kt

```diff
@@ -16,84 +16,87 @@ class BookService(private val bookRepository: BookRepository,
 
     fun createBook(bookDTO: BookDTO): BookDTO {
         val savedBookDTO = bookRepository.saveBookIfItNotExists(bookDTO)
-        if (savedBookDTO.currentPage > 0L) {
+        if (savedBookDTO.currentPage > 0) {
             saveBookUpdate(savedBookDTO)
         }
         return bookDTO
     }
 
     fun updateBook(bookUpdate: BookUpdateInputDTO, bookName: String): BookUpdatesFileDTO {
-        var response = BookUpdatesFileDTO()
-        var books = bookRepository.getBooks().books.filter { it.name != bookName }
-        var oldPage = Long.MAX_VALUE
-        val foundBook = bookRepository.getBookByName(bookName)
+        val foundBook = bookRepository.getBookByName(bookName) ?: return BookUpdatesFileDTO()
 
-        if (foundBook.name == bookName) {
-            foundBook.let {
-                oldPage = it.currentPage
-                it.currentPage = bookUpdate.currentPage
-                if (it.currentPage >= it.pagesTotal) {
-                    it.readTime = (Date().time - dateFormat.parse(it.dateStarted).time) / DIVISOR_FOR_DAY
-                }
-            }
-            books = mutableListOf(foundBook) + books.toMutableList()
+        if (foundBook.currentPage == foundBook.pagesTotal) {
+            foundBook.readTime = ((Date().time - dateFormat.parse(foundBook.dateStarted).time) / DIVISOR_FOR_DAY) + 1
+        }
+        val oldPage = foundBook.currentPage
+        foundBook.currentPage = bookUpdate.currentPage
+        val pagesRead = bookUpdate.currentPage.minus(oldPage)
+        if (pagesRead <= 0 || oldPage >= bookUpdate.currentPage || bookUpdate.currentPage > foundBook.pagesTotal) {
```

Comment:
```oldPage >= bookUpdate.currentPage``` is not needed if you already have ```pagesRead <= 0```

Reply:
I see. If the old page is higher than the new/current page, then pagesRead is never greater or equal 0. I will change it